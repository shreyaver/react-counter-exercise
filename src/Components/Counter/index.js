import React, { Component } from 'react';
import Button from '../Button';
import './Counter.css';

class Counter extends Component {
  state = {counter : this.props.counterVal};
  increment = () => {
    this.setState((state, props) => ({
      counter: this.state.counter + this.props.incrementVal
    }));
  }
  decrement = () => {
    this.setState((state, props) => ({
      counter: this.state.counter - this.props.decrementVal
    }));
  }
  
  render() {
      return (
      <div className = 'main'> 
      <p>Counter: {this.state.counter}</p>
      <Button caption = 'ADD' onClick = {this.increment}/>
      <Button caption = 'SUBTRACT' onClick = {this.decrement}/>
      </div>
      )
    }
  }
  export default Counter;
