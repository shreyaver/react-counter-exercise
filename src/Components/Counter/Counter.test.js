import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '.';
import { shallow } from 'enzyme'

describe('counter', () => {
  const initialValOfCounter = 1;
  const wrapper = shallow(<Counter counterVal={initialValOfCounter} incrementVal={2} decrementVal={3} />);
  beforeEach(() => {
    wrapper.instance().setState((state, props) => ({
      counter: initialValOfCounter
    }));
  });
  it('renders without crashing', () => {
    
    const counter = renderer.create(<Counter counterVal={1} incrementVal={2} decrementVal={3} />).toJSON();
    expect(counter).toMatchSnapshot();
  });
  it('sets initial state of counter to passed value', () => {
    
    expect(wrapper.instance().state.counter).toEqual(1);
  });
  it('increments counter by passed value', () => {
    
    wrapper.instance().increment();
    expect(wrapper.instance().state.counter).toEqual(3);
  });
  it('decrements counter by passed value', () => {
    
    wrapper.instance().decrement();
    expect(wrapper.instance().state.counter).toEqual(-2);
  });
});
