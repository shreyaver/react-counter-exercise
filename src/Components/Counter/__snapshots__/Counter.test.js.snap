// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter renders without crashing 1`] = `
<div
  className="main"
>
  <p>
    Counter: 
    1
  </p>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "color": "#fff",
      }
    }
    type="button"
  >
    ADD
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "color": "#fff",
      }
    }
    type="button"
  >
    SUBTRACT
  </button>
</div>
`;
